src/main/java/org/openstreetmap/atlas/checks/validation/intersections/EdgeCrossingEdgeCheck.java

```diff
@@ -105,22 +108,43 @@ public boolean validCheckForObject(final AtlasObject object)
         // edges from being flagged more than once.
         final Set<Edge> collectedEdges = new EdgeCrossingEdgeWalker((Edge) object,
                 this.getInvalidCrossingEdges()).collectEdges();
-
         if (collectedEdges.size() > 1)
         {
-            final CheckFlag newFlag = new CheckFlag(getTaskIdentifier(object));
-            this.markAsFlagged(object.getIdentifier());
-            newFlag.addObject(object);
-            newFlag.addInstruction(this.getLocalizedInstruction(0, object.getOsmIdentifier()));
-
-            collectedEdges.forEach(invalidEdge ->
+            final List<Tuple2<Edge, Set<Edge>>> edgeCrossPairs = collectedEdges
+                    .stream().filter(
+                            e -> e.getOsmIdentifier() != object.getOsmIdentifier())
+                    .map(e -> new Tuple2<>(e,
+                            new EdgeCrossingEdgeWalker(e, this.getInvalidCrossingEdges())
+                                    .collectEdges()))
+                    .collect(Collectors.toList());
+            edgeCrossPairs.add(new Tuple2<>((Edge) object, collectedEdges));
+            final Optional<Tuple2<Edge, Set<Edge>>> maxPair = edgeCrossPairs.stream().max(
+                    (entry1, entry2) -> Integer.compare(entry1._2().size(), entry2._2().size()));
+            if (maxPair.isPresent())
             {
-                this.markAsFlagged(invalidEdge.getIdentifier());
-                newFlag.addObject(invalidEdge);
-                newFlag.addInstruction(
-                        this.getLocalizedInstruction(1, invalidEdge.getOsmIdentifier()));
-            });
-            return Optional.of(newFlag);
+                final int maxSize = (maxPair.get()._2()).size();
+
+                // max edges object is not the one passed to this flag.
+                final List<Tuple2<Edge, Set<Edge>>> maxEdgePairs = edgeCrossPairs.stream()
+                        .filter(crossPair -> (crossPair._2()).size() == maxSize)
+                        .collect(Collectors.toList());
+                final Optional<Tuple2<Edge, Set<Edge>>> minIdentifierPair = maxEdgePairs.stream()
+                        .reduce((edge1, edge2) ->
+                        {
```

Comment:
Could you remove the return and curly brace to fix https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas-checks&open=AXBkpYtoPUfJDu18rpLu&pullRequest=253&resolved=false&types=CODE_SMELL